### PR TITLE
Replace manifest-tool with docker manifest command

### PR DESCRIPTION
**Pull request by mkumatag**
forked from https://github.com/kubernetes/kubernetes/pull/63453#issuecomment-395716068

### CODE REVIEW

**Comment by dims** [comment]
/milestone v1.11

We need to get this right before the release as we already pushed manifests using manifest tool for the test images at least once.

File: test/images/image-util.sh [hunk 1]
```diff
@@ -90,19 +90,23 @@ build() {
...
-    TAG=$(<${IMAGE}/VERSION)
     docker push ${REGISTRY}/${IMAGE}-${arch}:${TAG}
   done
 
-  # Make archs list into OS/architecture pair. Eg: 'amd64 ppc64le' to 'linux/amd64,linux/ppc64le'
-  archs=$(echo $archs | sed -e 's/[^ ]* */linux\/&/g' -e 's/ /,/g')
-  manifest-tool push from-args --platforms ${archs} --template ${REGISTRY}/${IMAGE}-ARCH:${TAG} --target ${REGISTRY}/${IMAGE}:${TAG}
+  # Make archs list into image manifest. Eg: 'amd64 ppc64le' to '${REGISTRY}/${IMAGE}-amd64:${TAG} ${REGISTRY}/${IMAGE}-ppc64le:${TAG}'
+  manifest=$(echo $archs | sed -e "s~[^ ]*~$REGISTRY\/$IMAGE\-&:$TAG~g")
+  docker manifest create --amend ${REGISTRY}/${IMAGE}:${TAG} ${manifest}
```
Comment (luxas) [inline test/images/image-util.sh]: nit: we might wanna do some version matching of the docker client first, so we know that it is new enough, or error out

**Review by luxas — APPROVED** [review]
/approve
/lgtm

**Pull request merged**